share/ios/Plugin/SharePlugin.swift

```diff
@@ -12,7 +12,11 @@ public class SharePlugin: CAPPlugin {
         }
 
         if let url = call.getString("url"), let urlObj = URL(string: url) {
-            items.append(urlObj)
+            if let imageData = try? Data(contentsOf: urlObj), let image = UIImage(data: imageData) {
```

Comment:
I'm concerned that this approach is too naive. We should not blindly try to load any URL without knowing what it is pointing to. If it's a giant video file, large pdf or simply on the other side of a slow network connection, it will block the thread while loading before it can even figure out if it's actually an image. And if the data is too large, it could run out of RAM and cause memory pressure problems for the whole app.

It either needs to be much smarter about checking the origin of the URL, guessing content type, and switching to a background thread when necessary, or it should put that burden on the app developer by requiring an additional parameter to know when to attempt this.

Reply:
what about doing it for local files only?

Reply:
I think only trying with file urls would be a lot safer and less likely to cause side effects. I'd also suggest checking the extension for image types.

That does mean that developers would need to download and save an image first if they wanted this behavior but as long as we document the rules I think that's a reasonable compromise.

Reply:
There are a lot of cases where the image doesn't have a file extension, so checking for it can be useless, but most times you can get the mime type from the url or from the data, so probably better to get it from there.